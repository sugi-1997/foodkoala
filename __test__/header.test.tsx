export {};

// import Header from './header';
// import renderer from 'react-test-renderer';
// import Link from 'next/link';
// import Image from 'next/image';
// import styles from 'styles/Header.module.css';
// import { fireEvent, render, screen } from '@testing-library/react';

// describe('Header', () => {
//   //関数以外のスナップショットを実施
//   it('renders header correctly when loginStatus is false', () => {
//     const loginStatus = 'false';
//     const logoutStatus = 'true';
//     const tree = renderer
//       .create(
//         <>
//           <header className={styles.header}>
//             <div className={styles.logo}>
//               <Link href="/">
//                 <Image
//                   src="/images/provisional_logo.png"
//                   alt="logo"
//                   width={35}
//                   height={35}
//                 />
//                 <span>Food Koala</span>
//               </Link>
//             </div>
//             {/* ここからハンバーガーメニュー */}
//             <div className={styles.hamburger}>
//               <input
//                 type="checkbox"
//                 id="menu_btn_check"
//                 className={styles.menu_btn_check}
//               />
//               <label
//                 htmlFor="menu_btn_check"
//                 className={styles.menu_btn}
//               >
//                 <span></span>
//               </label>
//               {/* ここまでハンバーガーメニュー */}
//               <nav className={styles.nav}>
//                 <ul>
//                   <li>
//                     <Link href="/">メニュー</Link>
//                   </li>
//                   <li>
//                     <Link href="/shop/list">ショップ</Link>
//                   </li>
//                   <li>
//                     <Link href="/shop/favorite">お気に入り</Link>
//                   </li>
//                   <li>
//                     <Link href="/order/list">注文リスト</Link>
//                   </li>
//                   <li>
//                     <Link href="/order/order_history">注文履歴</Link>
//                   </li>
//                   <li className={styles[loginStatus]}>
//                     <Link href="/login">ログイン</Link>
//                   </li>
//                   <li className={styles[logoutStatus]}>ログアウト</li>
//                 </ul>
//               </nav>
//             </div>
//           </header>
//         </>
//       )
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//   });

//   it('renders header correctly when loginStatus is true', () => {
//     const loginStatus = 'true';
//     const logoutStatus = 'false';
//     const tree = renderer
//       .create(
//         <>
//           <header className={styles.header}>
//             <div className={styles.logo}>
//               <Link href="/">
//                 <Image
//                   src="/images/provisional_logo.png"
//                   alt="logo"
//                   width={35}
//                   height={35}
//                 />
//                 <span>Food Koala</span>
//               </Link>
//             </div>
//             {/* ここからハンバーガーメニュー */}
//             <div className={styles.hamburger}>
//               <input
//                 type="checkbox"
//                 id="menu_btn_check"
//                 className={styles.menu_btn_check}
//               />
//               <label
//                 htmlFor="menu_btn_check"
//                 className={styles.menu_btn}
//               >
//                 <span></span>
//               </label>
//               {/* ここまでハンバーガーメニュー */}
//               <nav className={styles.nav}>
//                 <ul>
//                   <li>
//                     <Link href="/">メニュー</Link>
//                   </li>
//                   <li>
//                     <Link href="/shop/list">ショップ</Link>
//                   </li>
//                   <li>
//                     <Link href="/shop/favorite">お気に入り</Link>
//                   </li>
//                   <li>
//                     <Link href="/order/list">注文リスト</Link>
//                   </li>
//                   <li>
//                     <Link href="/order/order_history">注文履歴</Link>
//                   </li>
//                   <li className={styles[loginStatus]}>
//                     <Link href="/login">ログイン</Link>
//                   </li>
//                   <li className={styles[logoutStatus]}>ログアウト</li>
//                 </ul>
//               </nav>
//             </div>
//           </header>
//         </>
//       )
//       .toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
