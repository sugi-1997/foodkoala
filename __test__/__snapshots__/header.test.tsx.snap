// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders header correctly when loginStatus is false 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <img
        alt="logo"
        data-nimg="1"
        decoding="async"
        height={35}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fprovisional_logo.png&w=96&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fprovisional_logo.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fprovisional_logo.png&w=96&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={35}
      />
      <span>
        Food Koala
      </span>
    </a>
  </div>
  <div
    className="hamburger"
  >
    <input
      className="menu_btn_check"
      id="menu_btn_check"
      type="checkbox"
    />
    <label
      className="menu_btn"
      htmlFor="menu_btn_check"
    >
      <span />
    </label>
    <nav
      className="nav"
    >
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            メニュー
          </a>
        </li>
        <li>
          <a
            href="/shop/list"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            ショップ
          </a>
        </li>
        <li>
          <a
            href="/shop/favorite"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            お気に入り
          </a>
        </li>
        <li>
          <a
            href="/order/list"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            注文リスト
          </a>
        </li>
        <li>
          <a
            href="/order/order_history"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            注文履歴
          </a>
        </li>
        <li
          className="false"
        >
          <a
            href="/login"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            ログイン
          </a>
        </li>
        <li
          className="true"
        >
          ログアウト
        </li>
      </ul>
    </nav>
  </div>
</header>
`;

exports[`Header renders header correctly when loginStatus is true 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <img
        alt="logo"
        data-nimg="1"
        decoding="async"
        height={35}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimages%2Fprovisional_logo.png&w=96&q=75"
        srcSet="/_next/image?url=%2Fimages%2Fprovisional_logo.png&w=48&q=75 1x, /_next/image?url=%2Fimages%2Fprovisional_logo.png&w=96&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={35}
      />
      <span>
        Food Koala
      </span>
    </a>
  </div>
  <div
    className="hamburger"
  >
    <input
      className="menu_btn_check"
      id="menu_btn_check"
      type="checkbox"
    />
    <label
      className="menu_btn"
      htmlFor="menu_btn_check"
    >
      <span />
    </label>
    <nav
      className="nav"
    >
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            メニュー
          </a>
        </li>
        <li>
          <a
            href="/shop/list"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            ショップ
          </a>
        </li>
        <li>
          <a
            href="/shop/favorite"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            お気に入り
          </a>
        </li>
        <li>
          <a
            href="/order/list"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            注文リスト
          </a>
        </li>
        <li>
          <a
            href="/order/order_history"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            注文履歴
          </a>
        </li>
        <li
          className="true"
        >
          <a
            href="/login"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            ログイン
          </a>
        </li>
        <li
          className="false"
        >
          ログアウト
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
